src/abstractdataframe/abstractdataframe.jl

```diff
@@ -761,14 +761,28 @@ function completecases(df::AbstractDataFrame, col::Colon=:)
                             "data frame with no columns"))
     end
     res = trues(size(df, 1))
+    aux = BitVector(undef, size(df, 1))
     for i in 1:size(df, 2)
-        res .&= .!ismissing.(df[!, i])
+        v = df[!, i]
+        if Missing <: eltype(v)
+            # Disable fused broadcasting as it happens to be much slower
+            aux .= .!ismissing.(v)
+            res .&= aux
+        end
     end
-    res
+    return res
 end
 
-completecases(df::AbstractDataFrame, col::ColumnIndex) =
-    .!ismissing.(df[!, col])
+function completecases(df::AbstractDataFrame, col::ColumnIndex)
+    v = df[!, col]
+    if Missing <: eltype(v)
+        res = BitVector(undef, size(df, 1))
+        res .= .!ismissing.(v)
+        return res
```

Comment:
Why not just this?
```suggestion
        return .!ismissing.(v)
```

Reply:
because it is not type stable. I have just reversed this. The current design (with `res`) has no performance penalty, but pasess `@inferred`.

Reply:
That's surprising. How about just adding `::BitVector` in the function definition? That would make it clearer that the goal is to fix inference.

Reply:
I thought you would ask about it :).

The `::BitVector` annotation could potentially allocate once more by converting `Vector{Bool}` to `BitVector` while what I do should guarantee only that only one allocation happens because broadcasting assignment does `copyto!` of `Broadcasted` into target.

Here is an example:
```
julia> f1(x::AbstractVector)::BitVector = .!ismissing.(x)
f1 (generic function with 1 method)

julia> f2(x::AbstractVector) = (res = BitVector(undef, length(x)); res .= .!ismissing.(x); return x)
f2 (generic function with 1 method)

julia> using SparseArrays

julia> x = sparse(1:10^7);

julia> using BenchmarkTools

julia> f1(x::AbstractVector)::BitVector = .!ismissing.(x)
f1 (generic function with 1 method)

julia> @btime f1($x);
  568.808 ms (7 allocations: 87.02 MiB)

julia> @btime f2($x);
  525.133 ms (4 allocations: 1.20 MiB)
```